ksp_compiler3/ksp_compiler.py

```diff
@@ -1544,13 +1621,19 @@ def do_imports_and_convert_to_line_objects(self):
             else:
                 # if there is no persistence_changed callback then generate one
                 source = source + "\non persistence_changed\ncheckPrintFlag()\nend on\n"            
-
-
+                        
         self.lines = parse_lines_and_handle_imports(source,
                                                     read_file_function=self.read_file_func,
                                                     preprocessor_func=self.examine_pragmas)
         handle_conditional_lines(self.lines)
 
+        if import_nckp(source):
+            for line_obj in self.lines:
```

Comment:
This part could become a `strip_import_nckp_function_from_source` function

Reply:
done